test/endpoints/construction.test.ts

```diff
@@ -0,0 +1,206 @@
+import { describe, expect, inject, it } from "vitest";
+import { ConstructionApi, CurveType, SignatureType } from "../generated/api";
+import { hdNode, networkIdentifier, vthoAddress } from "../constants";
+import { abi, secp256k1 as Sign } from "thor-devkit";
+import { Erc20ABI } from "../abis";
+import { pollReceipt } from "../helpers/transactions/pollReceipt";
+
+describe("construction", async () => {
+  const client = new ConstructionApi(inject("rosettaURL"));
+
+  const delegator = hdNode.derive(2);
+  const origin = hdNode.derive(3);
+  const receiver = hdNode.derive(4);
+
+  it("should be able to fetch metadata", async () => {
+    const transferABI = new abi.Function(Erc20ABI.transfer);
+    const data = transferABI.encode(receiver.address, 100);
+
+    const metadata = await client.constructionMetadata({
+      networkIdentifier,
+      options: {
+        clauses: [
+          {
+            to: vthoAddress,
+            value: "0x0",
+            data,
+          },
+        ],
+      },
+    });
+
+    expect((metadata.body.metadata as any).chainTag).toBe(246);
+    expect(metadata.body.suggestedFee?.length).toBe(1);
+    expect(metadata.body.suggestedFee![0].value).toBe("-840000000000000000");
+    expect(metadata.body.suggestedFee![0].currency.symbol).toBe("VTHO");
+    expect(metadata.body.suggestedFee![0].currency.metadata).toEqual({
+      contractAddress: vthoAddress,
+    });
+  });
+
+  it("should be able to send a delegated transaction", async () => {
+    const preprocessBody = {
+      networkIdentifier: {
+        blockchain: "vechainthor",
+        network: "custom",
+      },
+      metadata: {
+        fee_delagator_account: delegator.address,
```

Comment:
Seems the readme documentation is outdated. It says the `FeeDelegation` operation is required, but it seems we should just pass this `metadata` option